src/test/test-copy.c

```diff
@@ -50,6 +51,146 @@ TEST(copy_file) {
         unlink(fn_copy);
 }
 
+TEST(copy_tree_replace_file) {
+        _cleanup_free_ char *buf = NULL;
+        char src[] = "/tmp/test-copy_file.XXXXXX";
+        char dest[] = "/tmp/test-copy_file.XXXXXX";
+
+        size_t sz = 0;
+        int fd;
+
+        fd = mkostemp_safe(src);
+        assert_se(fd >= 0);
+        close(fd);
+
+        fd = mkostemp_safe(dest);
+        assert_se(fd >= 0);
+        close(fd);
+
+        assert_se(write_string_file(src, "bar bar", WRITE_STRING_FILE_CREATE) == 0);
+        assert_se(write_string_file(dest, "foo foo foo", WRITE_STRING_FILE_CREATE) == 0);
+
+
+        assert_se(read_full_file(dest, &buf, &sz) == 0);
+        assert_se(streq(buf, "foo foo foo\n"));
+        assert_se(sz == 12);
```

Comment:
No need to check the sizes here, you can just pass NULL as the sz argument to read_full_file() here and elsewhere

Reply:
I left the size checks in because my original implementation mis-used `O_EXCL` which would have resulted in incorrect sizes (so my intention was for these checks to catch such an issue).
I copied it from a previous test as well so i assumed it was necessary, I can remove though

Reply:
Ahh, that makes sense, in that case, let's keep the size checks

Reply:
I thought it would be caught by streq :P but I assumed I was missing something and included it anyway